src/main/scala/com/johnsnowlabs/partition/Partition.scala

```diff
@@ -21,8 +21,97 @@ import org.apache.spark.sql.DataFrame
+  *   - timeout (HTML): Timeout in seconds for fetching remote HTML content.
+  *   - title_font_size (HTML, Excel): Minimum font size used to identify titles based on
+  *     formatting.
+  *   - include_page_breaks (Word, Excel): Whether to tag content with page break metadata.
+  *   - group_broken_paragraphs (Text): Whether to merge broken lines into full paragraphs using
+  *     heuristics.
+  *   - title_length_size (Text): Max character length used to qualify text blocks as titles.
+  *   - paragraph_split (Text): Regex to detect paragraph boundaries when grouping lines.
+  *   - short_line_word_threshold (Text): Max word count for a line to be considered short.
+  *   - threshold (Text): Ratio of empty lines used to switch between newline-based and paragraph
+  *     grouping.
+  *   - max_line_count (Text): Max lines evaluated when analyzing paragraph structure.
+  *   - include_slide_notes (PowerPoint): Whether to include speaker notes from slides as
+  *     narrative text.
+  *   - infer_table_structure (Word, Excel, PowerPoint): Generate full HTML table structure from
+  *     parsed table content.
+  *   - append_cells (Excel): Append all rows into a single content block instead of individual
+  *     elements.
+  *   - cell_separator (Excel): String used to join cell values in a row for text output.
+  *   - add_attachment_content (Email): Include text content of plain-text attachments in the
+  *     output.
+  *   - headers (HTML): This is used when a URL is provided, allowing you to set the necessary
+  *     headers for the request.
+  *
+  * ==Example 1 (Reading Text Files)==
+  * {{{
+  * val txtDirectory = "/content/txtfiles/reader/txt"
+  * val textDf = Partition(Map("content_type" -> "text/plain")).partition(txtDirectory)
+  * textDf.show()
+  *
+  * +--------------------+--------------------+
+  * |                path|                 txt|
+  * +--------------------+--------------------+
+  * |file:/content/txt...|[{Title, BIG DATA...|
+  * +--------------------+--------------------+
+  * }}}
+  *
+  * ==Example 2 (Reading Email Files)==
+  * {{{
+  * emailDirectory = "./email-files/test-several-attachments.eml"
+  * partitionDf = Partition(Map("content_type" -> "message/rfc822")).partition(emailDirectory)
+  * partitionDf.show()
+  * +--------------------+--------------------+
+  * |                path|               email|
+  * +--------------------+--------------------+
+  * |file:/content/ema...|[{Title, Test Sev...|
+  * +--------------------+--------------------+
+  * }}}
+  *
+  * ==Example 3 (Reading Webpages)==
+  * {{{
+  *   val htmlDf = Partition().partition("https://www.wikipedia.org")
+  *   htmlDf.show()
+  *
+  * +--------------------+--------------------+
+  * |                 url|                html|
+  * +--------------------+--------------------+
+  * |https://www.wikip...|[{Title, Wikipedi...|
+  * +--------------------+--------------------+
+  *
+  * }}}
+  * *
+  */
+
```

Comment:
It might be helpful to include a reference link to the Partition demo notebook that has examples with all available parameters: `examples/python/data-preprocessing/SparkNLP_Partition_Reader_Demo.ipynb`